Deployment (#6)
avantifellows/plio-analytics: docs/ENV.md

## Environment variables

> This guide explains all the available configurations in your `.env` file so that you can configure your environments accordingly. For more details, you can also check official documentation from Cube.js on [Environment Variables](https://cube.dev/docs/reference/environment-variables).

Reviewer: the link talks about this env variable:
```
CUBEJS_APP | An application ID used to uniquely identify the Cube.js deployment. Can be different for multitenant setups. Defaults to cubejs
```
Wouldn't we need this for our multitenancy part?

Author: I believe this is for multitenant set up of cubejs.

Used by CubeJS. It should be a secret value and shouldn't be added to git.
You can create a 32-digit secret key from this website: https://generate.plus/en/base64

> #### `CUBEJS_SCHEDULED_REFRESH_TIMER`

Reviewer: are we using pre-aggregations?

Reviewer: If this is for refreshing in-memory cache, what should be the value - just `true` or some set number of seconds - as a nice default?

Reviewer: how are we triggering scheduled refreshes? this is the recommended process: https://cube.dev/docs/config#options-reference-scheduled-refresh-timer

#### `APP_DASHBOARD_PORT`
Port on which you want the Cube.js Dashboard App to be created using the Developer Playground.

> 

Reviewer 2: In the `.env.example` I saw that you changed `APP_API_PORT` to `PORT`, but that change is not reflected here

### Redis
#### `CUBEJS_REDIS_URL`
Hostname of your Redis instance.

#### `CUBEJS_REDIS_PASSWORD`
Password of your Redis instance.

#### `CUBEJS_REDIS_TLS`

> To set TLS authentication between the app and Redis server.

Reviewer 2: Can you just add a word saying that it's boolean?
was not super clear to me at first glance

### Authentication
Before setting these variables, make sure you have gone through the [Authentication](./AUTHENTICATION.md) guide.

#### `CUBEJS_JWK_URL`

> This is the url to fetch the JSON wek keys details. Replace the `<AUTH0-SUBDOMAIN>` by the domain name for your account. The domain can be determined from `Your app > Settings > Domain`.

Reviewer 2: ```suggestion
This is the url to fetch the JSON web keys details. Replace the `<AUTH0-SUBDOMAIN>` by the domain name for your account. The domain can be determined from `Your app > Settings > Domain`.
```

#### `CUBEJS_JWT_AUDIENCE`
This is the unique identifier that was set when configuring the API in Auth0. Should be the url for your app.

#### `CUBEJS_JWT_ISSUER`
JWT issuer should be the domain for your Auth0 app. Replace the `<AUTH0-SUBDOMAIN>` by the domain name for your account. The domain can be determined from `Your app > Settings > Domain`.

#### `CUBEJS_JWT_ALGS`

> This is to remain RS256 to decode the access token generated by same algorithm.

Reviewer 2: ```suggestion
This should be set to RS256 to decode the access token generated by same algorithm.
```
